esrally/log.py

```diff
@@ -40,6 +43,61 @@ def configure_utc_formatter(*args, **kwargs):
     return formatter
 
 
+MutatorType = typing.Callable[[logging.LogRecord, typing.Dict[str, typing.Any]], None]
+
+
+class RallyEcsFormatter(ecs_logging.StdlibFormatter):
+    def __init__(
+        self,
+        *args: typing.Any,
+        mutators: typing.Optional[typing.List[MutatorType]] = None,
+        **kwargs: typing.Any,
+    ):
+        super().__init__(*args, **kwargs)
+        self.mutators = mutators or []
+
+    def format(self, record: logging.LogRecord) -> str:
+        log_dict = super().format_to_ecs(record)
+        self.apply_mutators(record, log_dict)
+        # ecs_logging._utils is a private module
+        # but we need to use it here to get the on spec JSON serialization
+        return ecs_logging._utils.json_dumps(log_dict)  # pylint: disable=protected-access
```

Comment:
I think we can avoid protected access like below. This follows [suggestion](https://github.com/elastic/ecs-logging-python/blob/a4ef6e1ffff4f41b0f66d52a8dc93e77f6f70ac3/ecs_logging/_stdlib.py#L168-L179) in `ecs_logging`.

```suggestion
    def format_to_ecs(self, record: logging.LogRecord) -> typing.Dict[str, typing.Any]:
        log_dict = super().format_to_ecs(record)
        self.apply_mutators(record, log_dict)
        return log_dict
```

Reply:
Of course I should have done that. Thanks. Not sure where my brain was when I was overriding that method.